src/Ampersand/Input/Archi/ArchiAnalyze.hs

```diff
@@ -0,0 +1,705 @@
+   archi2PContext :: (HasLogFunc env) => String -> RIO env (Guarded P_Context)
+   archi2PContext archiRepoFilename  -- e.g. "CArepository.xml"
+    = do -- hSetEncoding stdout utf8
+         archiRepo <- liftIO $ runX (processStraight archiRepoFilename)
+         let fst3 (x,_,_) = x
+         let elemLookup atom = (Map.lookup atom . Map.fromList . typeMap) archiRepo
+         let archiRepoWithProps = (grindArchi elemLookup.identifyProps []) archiRepo
+         let relPops = (filter (not.null.p_popps) . sortRelPops . map fst3) archiRepoWithProps
+         let cptPops = (filter (not.null.p_popas) . sortCptPops . map fst3) archiRepoWithProps
+         let elemCount archiConcept = (Map.lookup archiConcept . Map.fromList . atomCount . atomMap) relPops
+         let countPop pop = let sig = ((\(Just sgn)->sgn).p_mbSign.p_nmdr) pop in
+                            (show.length.p_popps) pop              ++"\t"++
+                            (p_nrnm.p_nmdr) pop                    ++"\t"++
+                            (p_cptnm.pSrc) sig                     ++"\t"++
+                            (show.length.eqCl ppLeft.p_popps) pop  ++"\t"++
+                            (showMaybeInt.elemCount.pSrc) sig      ++"\t"++
+                            (p_cptnm.pTgt) sig                     ++"\t"++
+                            (show.length.eqCl ppRight.p_popps) pop ++"\t"++
+                            (showMaybeInt.elemCount.pTgt) sig
+         writeFileUtf8 "ArchiCount.txt" $ T.pack
+          (   (L.intercalate "\n" . map countPop) relPops
+           <> (concat . map showArchiElems . atomCount . atomMap ) (relPops++cptPops ) 
+          )
+         logInfo ("ArchiCount.txt written")
+         return (mkArchiContext archiRepoWithProps)
+{- reminder:
+data P_Population
+  = P_RelPopu { p_src   :: Maybe String -- a separate src and tgt instead of "Maybe Sign", such that it is possible to specify only one of these.
+              , p_tgt   :: Maybe String -- these src and tgt must be more specific than the P_NamedRel
+              , p_orig  :: Origin       -- the origin
+              , p_nmdr  :: P_NamedRel   -- the named relation
+              , p_popps :: [PAtomPair]  -- the contents
+              }
+  | P_CptPopu { pos     :: Origin  -- the origin
+              , p_cpt   :: P_Concept  -- the concept the population belongs to
+              , p_popas :: [PAtomValue]  -- atoms in the initial population of that concept
+              }
+data P_NamedRel = PNamedRel { p_nrpos :: Origin, p_nrnm :: String, p_mbSign :: Maybe P_Sign }
+
+data P_Sign = P_Sign {pSrc :: P_Concept, pTgt :: P_Concept } deriving (Ord,Eq)
+data PAtomPair
+  = PPair { pppos :: Origin
+          , ppLeft  :: PAtomValue
+          , ppRight :: PAtomValue
+          } deriving (Eq,Ord,Show) -- Show is for QuickCheck error messages and/or input redundancy removal only!
+-}
```

Comment:
Please remove this, get used to use the Haddock generated index file for this kind of info

Reply:
done....